src/test/java/jenkins/plugins/logstash/LogstashWriterTest.java

```diff
@@ -266,14 +292,65 @@ public void writeBuildLogSuccess() throws Exception {
     // Verify results
     // No error output
     assertEquals("Results don't match", "", errorBuffer.toString());
-    verify(mockBuild).getLog(3);
+    String lines = JSONArray.fromObject(mockBuild.getLog(3)).toString();
+    verify(mockBuild, times(2)).getLog(3);
     verify(mockBuild).getCharset();
 
     verify(mockDao).buildPayload(eq(mockBuildData), eq("http://my-jenkins-url"), anyListOf(String.class));
-    verify(mockDao).push("{\"data\":{},\"message\":[\"test\"],\"source\":\"jenkins\",\"source_host\":\"http://my-jenkins-url\",\"@version\":1}");
+    verify(mockDao).push("{\"data\":{},\"message\":" + lines + ",\"source\":\"jenkins\",\"source_host\":\"http://my-jenkins-url\",\"@version\":1}");
     verify(mockBuildData).updateResult();
   }
 
+  @Test
+  public void writeProcessedSuccess() throws Exception {
+    String goodMsg = "test message";
+    String ignoredMsg = "ignored input";
+    String scriptString =
+      "if (payload['message']) {\n" +
+      "  if (payload['message'][0] =~ /" + ignoredMsg + "/) {\n" +
+      "    payload.clear()\n" +
```

Comment:
Another thing here is that in #40 it was requested to get *returned* value instead of setting `payload` variable. The premises were that users will usually deal with the messages. But later on it turned out users are more interested in modifying metadata. This now can be done with:
```
payload["my-new-field"] = "value1"
payload.discard("undesired-field")
```
And this would be the whole script. With return values one needs to be careful to add proper return statement. Especially when there are `if/else` conditions.

Additionally, if we accept return values, when doing `if/else` it is easy to forget and return `null` by the script. Then how should we handle the `null`? As to skip persisting or raise an error? If you tell me you want the `return` approach, I'll do, just tell me how to handle `null` returns.

Another implication is that if we do **not** use the `return` approach then accepting regular expression patterns or matchers will not be sane after that. But if users are mainly interested in metadata is this a problem?

Finally, we want this script functionality to be used in the post-build action, then how would a matcher work there? Does it make sense to apply matcher to each message in the `message` array?

I'm really in doubt what is better. I think I prefer a little bit the current approach as it makes harder to get it wrong IMO. But there are some upsides of the `return` approach as well. So sorry for asking so many times. If you still prefer `return` I'll go for the return.

Reply:
OK, lets do it they way you are proposing
I guess if we want concise message matcher we can expose it via a separate attribute etc.

Reply:
Thank you! will be going for a couple of days in the wild but plan to make this migrated to LogstashJobProperty next week.